From cb74b3113076d6e18c799a0c73978e1aea32bad0 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Fri, 6 Jul 2012 14:25:48 +0200
Subject: [PATCH 2/5] usbredirhost: Properly log an error when the initial
 device reset fails

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 TODO                        | 1 +
 usbredirhost/usbredirhost.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/TODO b/TODO
index e6715b1..2085006 100644
--- a/TODO
+++ b/TODO
@@ -4,3 +4,4 @@
 * cancel pending packets / active streams before reset?
 * add a queue_buf call to parser, use it in host to avoid memcpy of
   "in" bulk transfers
+* use libusb strerror equivalent instead of numeric error codes in logs
diff --git a/usbredirhost/usbredirhost.c b/usbredirhost/usbredirhost.c
index be75f87..2415c26 100644
--- a/usbredirhost/usbredirhost.c
+++ b/usbredirhost/usbredirhost.c
@@ -707,6 +707,7 @@ int usbredirhost_set_device(struct usbredirhost *host,
        so lets do that before hand */
     r = libusb_reset_device(host->handle);
     if (r != 0) {
+        ERROR("resetting device: %d", r);
         usbredirhost_clear_device(host);
         return libusb_status_or_error_to_redir_status(host, r);
     }
-- 
1.7.11.2

